lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -6,18 +6,39 @@
 library js_promise;
 
 import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class PromiseResolver<T extends Object?> {}
+
+extension PromiseResolverExtension<T extends Object?> on PromiseResolver<T> {
+  void resolve(T result) => js_util.callMethod(this, 'call', <Object>[this, if (result != null) result]);
+}
+
+@JS()
+@staticInterop
+class PromiseRejecter {}
+
+extension PromiseRejecterExtension on PromiseRejecter {
+  void reject(Object? error) => js_util.callMethod(this, 'call', <Object>[this, if (error != null) error]);
+}
 
 /// Type-safe JS Promises
 @JS('Promise')
 @staticInterop
-abstract class Promise<T> {
+abstract class Promise<T extends Object?> {
```

Comment:
Are any of these type parameters really adding value? It might make sense just to take `Object` values.